lib/Language/Slicer/Core.hs

```diff
@@ -293,6 +296,7 @@ data Trace = TExp (Syntax Trace)
                                             -- alternative in a case expression
            | TCaseR Trace (Maybe Var) Trace -- ^ Take "right constructor"
                                             -- alternative in a case expression
+           | TCaseExn Trace
```

Comment:
I would suggest to remove `TCaseExn` for now. Once we have a proper type checking algorithm we'll restore it and handle properly.